### DISCUSSION TITLE
Is there any way to achieve 'jump from method implementation to interface method definition'?

### DISCUSSION THREAD

**Discussion by JohnWickzzz** [ayamir/nvimdots#1115] [Q&A] [answered]
For now, I use lspsaga & glance plugins. I found that they offer the function 'Lspsaga goto_definition' and 'Glance implementations' can use for jump from definition to implementation. 
I read the plugin doc and search from the network, but don't find a convenient way jump from method implementation to interface method definition. Is there something I'm missing about the use of plugins? Or is there any other way to do it?

**Comment by charliie-dev** [comment]
I dont really get your point. What lang and lsp do u use?

Can u give us some screenshots or examples of what u want to achieve? After all, they're commands from different plugins, i'm not sure why u want do combine them.

**Reply by JohnWickzzz** [reply]
Sorry, forgot to say, I use java and jdtls. 

Like this, a service method in java project controller or service class.
![image](https://github.com/ayamir/nvimdots/assets/12469651/a9c8257f-57e1-43bf-bc59-9423a94d7301)

I use 'Lspsaga goto_definition' can jump to interface file.
![image](https://github.com/ayamir/nvimdots/assets/12469651/f4bb8a1a-b742-478a-92c1-7d8975978b9c)

Use 'Glance implementations' jump to implementation file.
![image](https://github.com/ayamir/nvimdots/assets/12469651/5e9e46e9-8eae-48a0-abbb-f68f08af8dcf)

If I'm in the impl class method, how can I jump to the interface method. 
（其实就是Java中的接口和实现，现在通过glance插件可以从方法调用处或者接口函数那里跳转到具体实现类，但是现在不知道如何在实现类跳转到接口函数）

**Reply by charliie-dev** [reply]
hmmm, this seems like a tricky one. Personally i had never wrote one line of java before.
I think the closest shot is to use `Lspsaga finder`, but i'm not sure if it meets your needs.

**Reply by JohnWickzzz** [reply]
Thanks for you help. I tried the` Lspsaga finder`, but it not show the interface method.

**Reply by ayamir** [reply]
TBH, neovim native lsp doesn't support such feature (jump from implementations to interfaces) and I think it's a fake demand. It can be easily achieved by `Lspsaga goto_definition` actually. Just move cursor to `FeedbackService` and `gD`.

**Comment by Jint-lzxy** [comment, marked as answer]
I think this really depends on the language server u're using, bc the formal specification doesn't **mandate** that servers must implement a `textDocument/switchSourceHeader` method. I know `clangd` supports this (extra) feature and [we have a mapping for it](https://github.com/ayamir/nvimdots/blob/7df41c45cb0c58b59d6ab994b4022b901e4ab832/lua/modules/configs/completion/servers/clangd.lua#L1-L23), but for Java, IMO u should contact ur LSP vendor or check their documentation to see if there's a similar feature available.

**Reply by JohnWickzzz** [reply]
😁 In jdtls latest version 1.31.0, this issue `Implement go to declaration` has been fixed. Just use `:lua vim.lsp.buf.declaration()` or create a shortkey.
